distsql/select_result.go

```diff
@@ -106,17 +106,21 @@ func (r *selectResult) fetch(ctx context.Context) {
 			return
```

Comment:
Should we check in here?

Reply:
Yes, we should. Per https://godoc.org/github.com/pingcap/tidb/kv#Response, resultSubset may be nil even if no error take place.
```go
	// Next returns a resultSubset from a single storage unit.
	// When full result set is returned, nil is returned.
	Next(ctx context.Context) (resultSubset ResultSubset, err error)
```